### PR TITLE
[Feature-WIP](CCR): Add ccr doris interface (WIP)

### PR DESCRIPTION
**Pull request by JackDrogon**
## Problem summary

WIP: support ccr feature, fe/be interface. Cleaning up commit information, gradually committing

FE在做写入的时候，Editlog会记录下来
修改了EditLog的实现，按照logId构建一个递增序列
底层BE在做publish时候会将rowset的meta信息写入一份到binlog作为前缀的rocksdb kv中，然后link 这次提交的commit的segment files到binlog文件夹下面
这样FE(meta)和BE(data)构建了一个逻辑上的binlog系列
外部通过抽取到另外的集群上去做回放
这块为了性能考虑做的是物理文件回放，之后考虑支持逻辑回放，物理回放性能会好一些，当然也有其他问题
fe那边根据commitseq作为游标，得到下一条提交的commit的meta信息
然后根据meta到be上去抽取真实的binlog文件

1371684133982_ pic

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by dataroaring — CHANGES_REQUESTED** [review]

**Review by dataroaring — CHANGES_REQUESTED** [review]

File: gensrc/thrift/FrontendService.thrift [hunk 1]
```diff
@@ -523,6 +523,33 @@ struct TLoadTxnBeginResult {
     4: optional i64 db_id
 }
 
+struct Coordinator {
+    1: required string host
+    2: required i32 port
+}
+
+struct TBeginTxnRequest {
+    1: optional string cluster
```
Comment (morningman) [inline gensrc/thrift/FrontendService.thrift:532]: All field should be optional. Do not use requried

**Review by morningman — CHANGES_REQUESTED** [review]

**Comment by JackDrogon** [comment]
run buildall

**Comment by dataroaring** [comment]
run buildall

**Comment by dataroaring** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.81 seconds
 stream load tsv:          449 seconds loaded 74807831229 Bytes, about 158 MB/s
 stream load json:         22 seconds loaded 2358488459 Bytes, about 102 MB/s
 stream load orc:          58 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          80.3 seconds inserted 10000000 Rows, about 124K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230526055808_clickbench_pr_151056.html

**Review by morningman — APPROVED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Comment by BiteTheDDDDt** [comment]
图片
maybe we need to adjust the log

**Pull request merged**